src/plugins/usage_collection/README.md

```diff
@@ -0,0 +1,139 @@
+# Kibana Usage Collection Service
+
+Usage Collection allows collecting usage data for other services to consume (telemetry and monitoring).
+To integrate with the telemetry services for usage collection of your feature, there are 2 steps:
+
+1. Create a usage collector.
+2. Register the usage collector.
+
+## Creating and Registering Usage Collector
+
+All you need to provide is a `type` for organizing your fields, and a `fetch` method for returning your usage data. Then you need to make the Telemetry service aware of the collector by registering it.
+
+### New Platform:
+
+1. Make sure `usageCollection` is in your optional Plugins:
+
+```json
+// plugin/kibana.json
+{
+  "id": "...",
+  "optionalPlugins": ["usageCollection"]
+}
+```
+
+2. Register Usage collector in the `setup` function:
+
+```ts
+// server/plugin.ts
+class Plugin {
+  setup(core, plugins) {
+    registerMyPluginUsageCollector(plugins.usageCollection);
+  }
+}
+```
+
+3. Creating and registering a Usage Collector. Ideally collectors would be defined in a separate director `server/collectors/register.ts`.
```

Comment:
You can add this to the conventions doc if you'd like: `src/core/CONVENTIONS.md`

Reply:
Didnt know about this file, thanks i'll do that